avocado/core/test.py

```diff
@@ -337,13 +429,74 @@ def datadir(self):
         host data files that are generic enough to be used for all tests
         contained in a given test file.
         """
+        # TODO: on removal of Test.datadir, move its code over to
+        # get_file_data_dir
         # Maximal allowed file name length is 255
         if (self.filename is not None and
                 len(os.path.basename(self.filename)) < 251):
             return self.filename + '.data'
         else:
             return None
 
+    def get_data(self, filename):
+        """
+        Returns the path of a data file, if one exists
+
+        This method respects the existence of directories in the
+        following order:
+
+        1) "$(TEST_FILENAME).data/$(TEST)/$(VARIANT-$HASH)", if a
+           variant exists
+        2) "$(TEST_FILENAME).data/$(TEST)/$(VARIANT)", if a
+           variant exists
+        3) "$(TEST_FILENAME).data/$(TEST)", if no variant is used
+        4) "$(TEST_FILENAME).data", as a fallback for files common
+           to all tests hosted on a single file
+
+        :param filename: the relative name of the file to be searched
+                         on the the possible data directories
+        :type filename: str
+        :returns: either the path to the data file or None
+        :rtype: str or None
+        """
+        log_fmt = 'DATA (filename=%s) => %s (%s)'
```

Comment:
Sure, people should store the `get_data` results so logging each query is good and should not pollute logs too much.